### PR TITLE
Updated peers list fix, light-roots updated when update is called

### LINKED ISSUE #8: update() does not update light-roots/ latest block hashes
Roadbump: sometimes a peer's latest SyncInfo.LatestBlockHeight is way beyond what the actual block height is! (obtained from `GET /status`)
```
I[2021-06-25|18:13:48.031] Chose arbitrary peer to provide latest block height peer=http://139.59.70.20:26657 height=6710547
I[2021-06-25|18:13:48.031] Updating light roots from                    peer=http://hub.technofractal.com:26657
E[2021-06-25|18:13:48.377] error getting light roots from               peer=http://hub.technofractal.com:26657 error="RPC error -32603 - Internal error: height 6710547 must be less than or equal to the current blockchain height 6422185"
```

### PR DESCRIPTION
**Pull request by randomshinichi**
Closes #8

### CODE REVIEW

File: pkg/node/node.go [hunk 1]
```diff
@@ -179,13 +205,104 @@ func RefreshPeers(peers map[string]*Peer, logger log.Logger) (err error) {
...
+	// Transform map into list of peers
+	vs := []*Peer{}
+	for _, v := range peers {
+		vs = append(vs, v)
+	}
+
+	for i := 0; i < len(vs); i++ {
+		latestBlockHeight, err = mustGetLatestBlockHeight(vs[i], chainID, logger)
+		if err == nil {
+			return latestBlockHeight, err
```
Comment (alexanderbez) [inline pkg/node/node.go]:
```suggestion
		if err == nil {
			return latestBlockHeight, nil
```

File: pkg/node/node.go [hunk 2]
```diff
@@ -179,13 +205,104 @@ func RefreshPeers(peers map[string]*Peer, logger log.Logger) (err error) {
...
+		} else {
+			logger.Debug("Peer could not tell us the latest block height", "peer", vs[i].Address)
+		}
```
Comment (alexanderbez) [inline pkg/node/node.go]:
```suggestion
		} 
		
		logger.Debug("Peer could not tell us the latest block height", "peer", vs[i].Address)
```

File: pkg/node/node.go [hunk 3]
```diff
@@ -179,13 +205,104 @@ func RefreshPeers(peers map[string]*Peer, logger log.Logger) (err error) {
...
+}
+
+func mustGetLatestBlockHeight(peer *Peer, chainID string, logger log.Logger) (latestBlockHeight int64, err error) {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
+	defer cancel()
+
+	client, e := Client(peer.Address)
+	if e != nil {
+		logger.Error("error creating tendermint client: %s", e)
```
Comment (alexanderbez) [inline pkg/node/node.go]:
```suggestion
	client, err := Client(peer.Address)
	if err != nil {
		logger.Error("error creating tendermint client: %s", err)
```

File: pkg/node/node.go [hunk 4]
```diff
@@ -179,13 +205,104 @@ func RefreshPeers(peers map[string]*Peer, logger log.Logger) (err error) {
...
+		switch {
+		case err != nil:
+			logger.Error("error fetching client status: %s", "error", err)
+			return 0, err
+		case stat.NodeInfo.Network != chainID:
+			logger.Error("node(%s) is on chain(%s) not configured chain(%s)", peer.Address, stat.NodeInfo.Network, chainID)
+			return 0, err
+		}
+		if !stat.SyncInfo.CatchingUp {
+			return stat.SyncInfo.LatestBlockHeight, err
```
Comment (alexanderbez) [inline pkg/node/node.go]:
```suggestion
		if !stat.SyncInfo.CatchingUp {
			return stat.SyncInfo.LatestBlockHeight, nil
```

File: pkg/node/node.go [hunk 5]
```diff
@@ -179,13 +205,104 @@ func RefreshPeers(peers map[string]*Peer, logger log.Logger) (err error) {
...
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 2*time.Second)
+	defer cancel()
+
+	client, e := Client(peer.Address)
+	if e != nil {
+		logger.Error("error creating tendermint client: %s", e)
+	}
+
+	for {
```
Comment (alexanderbez) [inline pkg/node/node.go]: nit: I'd just use a ticker here instead of manual `Sleep` calls 👍

File: pkg/node/node.go [hunk 6]
```diff
@@ -179,13 +205,104 @@ func RefreshPeers(peers map[string]*Peer, logger log.Logger) (err error) {
...
+			commit, err = client.Commit(ctx, &h)
+			if err != nil {
+				logger.Error("error getting light roots from", "peer", peer.Address, "error", err)
+				return
+			}
+			lr := NewLightRoot(commit.SignedHeader)
+			lrr.AddResult(peer.ID, lr)
+			logger.Debug("Updated light roots from", "peer", peer.Address, "peerID", peer.ID, "lightroot", lr)
+		}(peer, nlr, logger)
+		wg.Add(1)
```
Comment (alexanderbez) [inline pkg/node/node.go]: Call `wg.Add(1)` _before_ invoking the go routine.

File: pkg/node/node.go [hunk 7]
```diff
@@ -179,13 +205,104 @@ func RefreshPeers(peers map[string]*Peer, logger log.Logger) (err error) {
...
+	for _, peer := range peers {
+		go func(peer *Peer, lrr *LightRootResults, logger log.Logger) {
+			defer wg.Done()
+			ctx := context.Background()
+			ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+			defer cancel()
+
+			client, e := Client(peer.Address)
+			if e != nil {
+				logger.Error("error creating tendermint client: %s", e)
```
Comment (alexanderbez) [inline pkg/node/node.go:293]: shouldn't we return here?

File: pkg/node/node.go (hunk 5 above)
Comment (randomshinichi) [reply pkg/node/node.go]: Interesting, didn't know about tickers before. How about now?

**Pull request merged**